ext/drmgr/drmgr.dox

```diff
@@ -104,6 +105,37 @@ for each instruction, each registered component is invoked.  This simplifies
 register allocation (register allocation is provided by a separate
 Extension \p drreg).
 
+\subsection sec_drmgr_emulation Emulation-Aware Instrumentation
+
+Support for one client library emulating or refactoring application code
+while another observes application behavior is provided through emulation
+marking and reading support.
+
+An emulation-creating library uses drmgr_insert_emulation_start() and
+drmgr_insert_emulation_end() to mark the emulation region and make a copy
+of the original instruction being replaced.
+
+An emulation-aware observational client then wants to instrument the
```

Comment:
Emulation unaware observational clients will risk correctness issues right? Just thinking whether being emulation-aware is optional or not.

Reply:
Correctness in capturing accurate program behavior: yes.  As the commit message shows, memtrace_simple recorded a 1-byte "movs" instruction instead of a 2-byte "rep movs".  I added another sentence on this.